crates/jstzd/src/protocol/mod.rs

```diff
@@ -0,0 +1,186 @@
+use rust_embed::Embed;
+use serde_json::Value;
+use std::fmt::Display;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum ProtocolConstants {
+    Sandbox,
+    #[cfg(test)]
+    TestConstants,
```

Comment:
Looking at the tests, if we just set protocol to alpha and constant to sandbox, it will automatically load base parameters from the `resource/protocol_parameters/sandbox/ProtoAlpha...` which is a sufficient test. `TestConstants` and `TestVersion` could be removed.

Reply:
I added `TestConstants` and `TestVersion` to check that the values were actually updated with the setters. The default values for these fields in the builder are `Sandbox` and `Alpha`, so if we set these values but the underlying logic somehow is wrong, we won't be able to detect it by looking for `Sandbox` and `Alpha`. We wouldn't need this if there are more than one possible enum values, but we don't at this moment. If you still think this is not really necessary, I can remove it.

Reply:
Oh I see, `new` calls `default`. Hmm should `new` create an empty Builder with optional fields? That's more conventional as a builder.

Either way, I don't think the test enums are necessary.

Reply:
I was thinking of setting the default values easily so didn't use optional fields. I've changed the logic here so that we can still kind of test default values with optional fields and without the test enums.